### PR TITLE
Fix when targeting via pillar with Salt syndic

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Fixing a weird edge case when using salt syndics and targeting via pillar.  Without this fix the master of masters ends up in an infinite loop since the data returned from the minions is differently structured than if a sync was not in use.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
When using a Salt syndic setup and targeting minions by Pillar values from the master of masters, the process would end up in an infinite loop since the data returned from the minions was different than what the previous logic was expected.

### New Behavior
Adds an additional case looking for empty return data to allow the processes to properly exit once the minions have returned.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/client/__init__.py [hunk 1]
```diff
@@ -1243,6 +1243,10 @@ def get_iter_returns(
                 if raw['data']['return'] == {}:
                     continue
 
+                if 'return' in raw['data']['return'] and \
+                    raw['data']['return']['return'] == {}
```
Comment (cachedout) [inline salt/client/__init__.py]: I believe you are missing a colon at the end of this conditional.

**Review by cachedout — COMMENTED** [review]
This appears to have a syntax error.

File: salt/client/__init__.py (hunk 1 above)
Comment (garethgreenaway) [inline salt/client/__init__.py]: @cachedout Damn!  Totally missed that!  Pushing an update now.

Comment (cachedout) [inline salt/client/__init__.py]: No worries. That's why we do reviews. :] Thanks, @garethgreenaway

**Comment by garethgreenaway** [comment]
@cachedout I noticed this problem in other older branches and can prepare PRs against those as well if you want, unless it's easier to merge down.

**Comment by cachedout** [comment]
Just give us one for 2016.3 and it will make its way forward. Thanks.

**Comment by garethgreenaway** [comment]
@cachedout done.

**Pull request merged**